staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/types.go

```diff
@@ -139,3 +140,91 @@ type CustomResourceDefinitionList struct {
 	// Items individual CustomResourceDefinitions
 	Items []CustomResourceDefinition
 }
+
+// CustomResourceValidation is a list of validation methods for CustomResources.
+// Can add other validation methods later if needed.
+type CustomResourceValidation struct {
+	// OpenAPISpecV2 is the OpenAPI v2 schema to be validated against.
+	OpenAPISpecV2 *JSONSchemaProps
```

Comment:
As @enisoc suggested (https://github.com/kubernetes/kubernetes/issues/49879#issuecomment-321407614), OpenAPIV2Schema is a better name.